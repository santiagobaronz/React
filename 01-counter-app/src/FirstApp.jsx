import PropTypes from 'prop-types'

export const FirstApp = ({ title, subtitle }) => {
    
    return(
        <>
            <h1 data-testid='test-title'>{ title }</h1>
            <p>{ subtitle }</p>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    title: 'No hay titulo',
    subtitle: 'No hay subtitulo'
}