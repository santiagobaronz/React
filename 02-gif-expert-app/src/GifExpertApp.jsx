import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['OnePunch'])

    const onAddCategory = ( newCategory ) => {
        if(categories.includes(newCategory)) return;
        setCategories([newCategory, ...categories,])
    }

  return (
    <>
        <h1>GiftExpertApp</h1>
        <AddCategory onNewValue={ onAddCategory } />
        { categories.map( (category) => <GifGrid key={category} category={category}/>)}
    </>
  )
}
