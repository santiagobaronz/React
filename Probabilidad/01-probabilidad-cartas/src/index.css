@import url("https://fonts.googleapis.com/css2?family=Poppins&display=swap");
@import url("https://cdn.jsdelivr.net/npm/bootstrap-icons@1.10.3/font/bootstrap-icons.css");

@tailwind base;
@tailwind components;
@tailwind utilities;

body::-webkit-scrollbar{
    width: 7px;
    background-color: transparent;
}
body::-webkit-scrollbar-track{
    background-color: #000000;
}

body::-webkit-scrollbar-thumb{
    background-color: #252525;
    border-radius: 5px;
}

.span-info{
	padding: 8px 30px;
	color: #6e727d;
	border: 1px solid #707378;
	margin: auto;
	text-align: center;
	width: max-content;
}

input[type="number"]::-webkit-outer-spin-button,
input[type="number"]::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

input[type="number"] {
  -moz-appearance: textfield;
  appearance: none;
}