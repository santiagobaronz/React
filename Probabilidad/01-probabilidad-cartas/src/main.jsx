import React from 'react'
import ReactDOM from 'react-dom/client'

import { HomePage } from './components/HomePage'

import './index.css'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
	<HomePage/>
  </React.StrictMode>,
)
