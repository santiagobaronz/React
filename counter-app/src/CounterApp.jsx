import PropTypes from 'prop-types'

export const CounterApp = ({ value }) => {
    
    const handleAdd = () => {
        console.log('+1')
    }
    
    return (
        <>
            <h1>CounterApp</h1>
            <h2>{ value }</h2>

            <button onClick={ handleAdd }>
                +1
            </button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}