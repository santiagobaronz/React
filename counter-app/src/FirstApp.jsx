const newMessage = 'Soy Santiago';

export const FirstApp = () => {
    return(
        <>
            <h1>{ newMessage }</h1>
            <p>Soy un parrafo</p>
        </>
    )
}