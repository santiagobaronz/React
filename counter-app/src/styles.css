html, body {
    background-color: #21232A;
    color: white;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button{
    padding: 15px 30px;
    border-radius: 10px;
    border: none;
    background-color: cadetblue;
    color: #FFF;
    font-size: 20px;
    margin-right: 20px;
}