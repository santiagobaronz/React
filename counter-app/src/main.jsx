import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirstApp } from './FirstApp';
import { HelloWorldApp } from './HelloWorldApp';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <FirstApp></FirstApp>
    </React.StrictMode>
)