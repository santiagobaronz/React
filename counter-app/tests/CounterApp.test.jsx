import { render, screen } from "@testing-library/react"
import { CounterApp } from "../src/CounterApp";

describe('Pruebas con el <CounterApp />', () => { 

    const value = 10;
    
    test('Debe hacer match con el snapshot', () => { 
        const { container } = render(<CounterApp value={value}/>)
        expect(container).toMatchSnapshot()
     })

     test('Debe mostrar el valor inicial de 100', () => { 
        render(<CounterApp value={100}/>)
        expect(screen.getByText(100)).toBeTruthy()
     })


 })