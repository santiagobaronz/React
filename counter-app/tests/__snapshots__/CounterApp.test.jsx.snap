// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el <CounterApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
